SCA: Remove the old "three_d_secure" logic

RadoslavGeorgiev opened the pull request:
Fixes #788.

#### Changes proposed in this Pull Request:

Cleanup of 3D Secure-related code, as PaymentIntents will handle authentication within an overlay on the checkout page.

DanReyLop commented:
> The _Require 3D Secure when applicable_ setting is not removed yet, as it might be still possible to force 3DS when it is `optional` for a card.

You _did_ remove it in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/795/commits/d529140d8cace17d0a0aee64dd8792b52107c3fe though.

I'm not sure what's the best way to handle this. What about:
- Remove the "Require 3DS when applicable" UI option.
- If the `three_d_secure` option is set to `true`, that means that this is an existing merchant who updated the extension and had higher-than-normal standards for security. In that case, show a dismissable admin notice or text in the settings page pointing to the "Radar rules" Stripe documentation.

Using the potential future API to set Radar rules from the extension is a bit overkill IMHO, and wouldn't be useful for 99% of merchants.

What do you think?

RadoslavGeorgiev commented:
My bad, I did not double-check the full description of the PR.

> The _Require 3D Secure when applicable_ setting is not removed yet, as it might be still possible to force 3DS when it is `optional` for a card. I started the discussion about this in [p1550750838004500-slack-woo-stripe-external](https://href.li/?https://a8c.slack.com/archives/woo-stripe-external/p1550750838004500)

I added this paragraph temporarily, while the topic is being discussed and left the PR as `[Status] In Progress`. Today I actually removed the setting and changed the status, but forgot to remove the paragraph from the description 🤦‍♂️ 

> Using the potential future API to set Radar rules from the extension is a bit overkill IMHO, and wouldn't be useful for 99% of merchants.

I fully agree, this would be an unnecessary amount of work, both in terms of initial coding and support afterwards.

> If the `three_d_secure` option is set to `true`, that means that this is an existing merchant who updated the extension and had higher-than-normal standards for security. In that case, show a dismissable admin notice or text in the settings page pointing to the "Radar rules" Stripe documentation.

I was tempted to suggest a message as the replacement of the field in the settings page, but I can see how that might be confusing to most merchants. The dismissable notice makes much more sense. Should we do it here or in another PR?

DanReyLop commented:
> Should we do it here or in another PR?

I don't mind. This PR is simple and the notice is related with this change, so it would fit here.

> I was tempted to suggest a message as the replacement of the field in the settings page, but I can see how that might be confusing to most merchants.

A message as the replacement of the field would be good too, since most merchants have "admin notice blindness". I'm gonna do a quick poll cc/ @woocommerce/hydra 

We are going to remove this setting since it doesn't really apply anymore:
![screenshot 2019-02-25 at 13 31 32](https://user-images.githubusercontent.com/1715800/53340775-b0794980-3901-11e9-95d1-f021b27cd192.png)

Instead, the rules as to when to require 3DS for a transaction can be ser in the [Stripe merchant dashboard](https://stripe.com/docs/payments/dynamic-3ds).

What should we do to direct merchants to Stripe Radar? React to this comment as appropiate.
- 😕 Don't do anything. Assume that a merchant who cares about security will find his way to the Stripe Radar dashboard.
- 👀 For merchants that had the option enabled, show them a dismissable Admin notice pointing them to the Dynamic 3DS doc (or any doc selated to that).
- 🚀 For merchants that had the option enabled, replace the option with a text blurb pointing them to the Dynamic 3DS doc.
- ❤️ For all merchants, replace the option with a text blurb pointing them to the Dynamic 3DS doc.

RadoslavGeorgiev commented:
@DanReyLop the notice is now in place ([#](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/795/commits/6d48b1b0e3dd2d07ee4929b92899258c6ccc2bd9)).

DanReyLop commented on assets/js/stripe-admin.js:
```diff
@@ -59,6 +59,21 @@ jQuery( function( $ ) {
+
+				$setting.find( '.notice-dismiss' ).on( 'click.wc-stripe-dismiss-notice', function() {
+					$.ajax( {
+						type: 'post',
```
Using a `HEAD` request with the nonce as a querystring parameter would be much more efficient.

DanReyLop commented on includes/class-wc-gateway-stripe.php:
```diff
@@ -961,4 +969,35 @@ public function get_intent_object( $intent_id = '' ) {
+			<p>
+				<?php
+				// ToDo: Replace this URL with a Stripe link or a WooCommerce article.
+				$url     = 'http://example.com';
```
I think https://stripe.com/docs/payments/dynamic-3ds would be a good URL to point to.

DanReyLop commented on includes/class-wc-gateway-stripe.php:
```diff
@@ -961,4 +969,35 @@ public function get_intent_object( $intent_id = '' ) {
+				<?php
+				// ToDo: Replace this URL with a Stripe link or a WooCommerce article.
+				$url     = 'http://example.com';
+				$message = __( 'We see that you had the "Require 3D secure when applicable" setting turned on. This setting is not available here anymore, because it is now replaced by Stripe Radar. You can learn more about it <a href="%s">here</a>.', 'woocommerce-gateway-stripe' );
```
You should mention Stripe in this message.
